### DISCUSSION TITLE
Reducing burden of correcting user expectations

### DISCUSSION THREAD

**Discussion by Danack** [community/community#5709] [Projects and Issues]
There is an increasing problem with end-users of Open Source projects having expectations for support that are appropriate for commercial software projects, not for stuff done for free.

I think part of the problem comes the way that the Github interface makes the whole site feel like it belongs to the end-user. It's not at all obvious that when you are on a repo that belongs to someone else, you are 'on their turf' and should behave differently to how you do 'at home'.

One example of this is when people open an issue, they think they are creating something that 'belongs to them' and get annoyed if the issue is closed without their problem being resolved.

I'm not going to link to any of the drama I've seen, but some examples are:

> "I do not need to convince anyone that this is worth implementing to keep this ticket open. It is whoever wants to close this who needs to explain why this would be invalid."

> "Closing a ticket means that the reported issue is solved; it does not merely express that someone chose not to fix it."

> "Thanks, but issues are closed when they are solved. It is OK not to intend to solve all issues, you can just leave the tickets as they are."

> "A ticket is not closed just because a developer does not intend to work on it."

Currently, each individual open source project has to explain to every new user who expects their issues to belong to the user, that actually the Github issue tracker is there for the benefit of the project, and doesn't belong to the user.

This is emotionally tiring, as every maintainer has to figure how to do this individually, and it often results in abusive language from the end-user as from the perspective of the end-user, the maintainer is acting in an unexpected manner.

It would make life easier for open source projects, if there was an official Github page that listed a set of expectations including:

* The issue tracker is for the benefit of the project. Even though you as an individual user might want an issue left open, projects are going to handle issues in their own best interest.

* If one of your issues is closed, opening another issue immediately is an annoying thing to do.

* Some variant of words written by [Rich Hickey](https://gist.github.com/richhickey/1563cddea1002958f96e7ba9519972d9) "As a user of something open source you are not thereby entitled to anything at all. You are not entitled to contribute. You are not entitled to features. You are not entitled to the attention of others. You are not entitled to having value attached to your complaints. You are not entitled to this explanation."

It wouldn't stop the end-user behaviour, but it would reduce the burden on open source maintainers, as they wouldn't have to make the arguments themselves, they would be able to just point to the official Github guidelines.

cheers
Dan
Ack

**Comment by pblitz** [comment]
One idea I've had for this is a default present issue state of " won't fix" as closed, the same thing that Jira has. 
Puts the discussion with the end users on a different state. It's not closed per se.

**Comment by Pen-y-Fan** [comment]
[Caleb Porzio](https://github.com/calebporzio) had released two podcasts in June, which agree with you, he states that "Github issues are broken", he has moved them to discussions, on [Alpine.js](https://github.com/alpinejs/alpine) and [Livewire](https://github.com/livewire/livewire):

If you have twenty minutes free, there are well worth a listen:

- [I killed the issues tab on the Alpine repo](https://share.transistor.fm/s/fc094ed5) -  let's talk about why. (10:14)
- [Still Loving GitHub Discussions](https://share.transistor.fm/s/c8c3048a) - Issues are broken. Discussions for LYYFEEEE. (9:39)

You can also see both projects have active discussions and look like they are working well.

**Reply by Danack** [reply]
> a default present issue state of " won't fix" as closed

That would change the person from saying "why was the issue closed" to "you don't care about this project, you're just going to leave valid issues as won't fix!!1!".

Some people will always use as emotionally strong language as they think they can get away with, to try to get their way.

**Reply by Danack** [reply]
And after listening to those podcasts, the situation is different for those projects than mine:

i) Having stuff be labelled as 'discussion' would probably bring a lot more stuff to deal with, as more people would be asking "how can I implement this?". Currently people mostly don't do that, as asking for help isn't an 'issue'.

ii) It's feasible for that maintainer to expect users who care enough to submit pull-requests, as that project is written in the same language that it's used in. The stuff I maintain is used by PHP programmers, but is written in C. Probably less than 5% of the users of my software would be able to submit PRs for it.

I'm possibly also slightly better at ignoring a large number of open issues.

**Comment by Pen-y-Fan** [comment]
If discussions are a solution, particularly for inexperienced users or people who don't get open-source:

- for new open-source projects, should discussions be enabled by default? 
- should issues be created only via a discussion? 
- should the tab order be changed so discussions are first? 
- what would be the easiest way to implement this for existing projects?
- would the above help lower the burden?
Edited: typo

**Reply by Ocramius** [reply]
Perhaps, by default, replace "new issue" with "new discussion" in repositories view, where it is possible, via alternative selection (dropdown near the button, perhaps) to create a "new issue".

This means:

 * by default, discussions will be created
 * users that know about the dropdown (more experienced) will create issues

It's a decent level of pre-filtering, I suppose.

**Reply by Danack** [reply]
> should issues be created only via a discussion?

No. Maybe other projects will want that, and it could be configurable for them, but I personally strongly dislike discussions as I find they actually make more work for me, and make it harder for people to open issues.

> would the above help lower the burden?

A small percent, and perhaps more for some projects, as there are some people who can be guided into better etiquette.

But there are also a percentage of people, who reason about the world quite differently, who even after someone takes the time to explain that their behaviour is making a burden for other people, continue to behave in that manner and double-down on "if you can't take feedback, then you should stop doing open source". 

The aim for my request is to make it easier for maintainers to end a conversation that isn't being productive for them, rather than each individual maintainer having to figure that out from scratch. Unproductive conversations will happen both through discussions and issues.

**Comment by jrfnl** [comment]
As for the "opening a duplicate issue straight after the original issue was closed part":
For some projects I have the feeling that about half the issues are duplicates anyway as users don't search before opening an issue.

Might be an idea (both for issues as well as discussions) to have an box along the lines of the below between the issue description box and the "submit" button which does an intelligent search across both open as well as closed issues, discussions and PRs:
> The below tickets look related, are you sure you are not opening a duplicate issue ?
> * Linked issue title 
> * Linked issue title 
> * Linked issue title

**Comment by evi-liu** [comment]
👋 Hey folks, jumping in here as the Discussions product manager. Thank you all for such a thoughtful and detailed conversation about this, and I empathize greatly with the burden maintainers go through when managing their communities. A couple of thoughts on my end from a Discussions standpoint:

- We do hope Discussions is the natural place and gateway for new users to get started and ask any of their questions. We recommend using Issue templates to redirect any off-topic Issues to Discussions. Here's an [example ](https://github.com/vercel/next.js/issues/new/choose)of how Next.js has set up their Issue template to redirect all questions to Discussions. 
- That being said, we have heard that Discussions can increase maintainer burden for some communities as this presents another space for them to moderate. With that in mind, we intentionally decided NOT to enable Discussions automatically for every repository. But of course, we highly encourage enabling Discussions to separate Issues out from questions/feedback/conversations if you do receive a large influx of these types of issues. 
- I agree there's a lot we can do to improve discoverability of Discussions for new users AND setting user expectations of engagement. You'll see some changes to the overall GitHub repo navigation in the coming months that will help facilitate this. In addition, we're planning to beef up our documentation in the coming months to outline the difference in usage and expectations between Issues and Discussions.

**Reply by Danack** [reply]
Hi @evi-liu,

Thanks for the answer (and apologies for slow response), but I think we're talking past each other slightly here.

For discussions, success would look like lots of discussions. For maintainers dealing with difficult users, fewer (or no) discussions would be the success case.

It would be inappropriate for me to go into too much depth about an individuals behaviour in a open issue tracker/discussion. Is there a not as public comms channel I could use to send you some more details of my experiences that lead me to open this issue? btw my email is in my profile.

**Comment by omerbensaadon** [comment]
@Danack we're introduced a new closed state called "Close as not planned" does that address your core concern here? 

https://docs.github.com/en/issues/tracking-your-work-with-issues/closing-an-issue

**Reply by Danack** [reply]
> "Close as not planned" does that address your core concern here?

No.

@omerbensaadon There are some details that I would like to communicate with you, that are probably not appropriate to be published publicly. Is there a way for me to send you details of a user performing a "distributed campaign of harassment of open source projects" ?
